apps/api-documenter/src/documenters/ExperimentalYamlDocumenter.ts

```diff
@@ -118,11 +101,57 @@ export class ExperimentalYamlDocumenter extends YamlDocumenter {
     }
   }
 
+  /**
+   * Filtering out the api-item by inlineTags or category name presence in the item name.
+   */
+  private _filterItem(apiItem: ApiItem, tocItem: IYamlTocItem): void {
+    const { categoryInlineTag, categorizeByName } = this._config;
+    const { name: itemName } = tocItem;
+    let filtered: boolean = false;
+
+    // First we attempt to filter by inline tag if provided.
+    if (apiItem instanceof ApiDocumentedItem) {
+      const docInlineTag: DocInlineTag | undefined =
+        categoryInlineTag
+          ? this._findInlineTagByName(categoryInlineTag, apiItem.tsdocComment)
+          : undefined;
+
+      const tagContent: string | undefined =
+        docInlineTag && docInlineTag.tagContent && docInlineTag.tagContent.trim();
+
+      if (tagContent && this._tocPointerMap[tagContent]) {
+        // null assertion used because when pointer map was created we checked for presence of empty `items` array
+        this._tocPointerMap[tagContent].items!.push(tocItem);
+        filtered = true;
+      }
+    }
+
+    // If not filtered by inline tag and `categorizeByName` config is enabled attempt to filter it by category name.
+    if (!filtered && categorizeByName) {
+      const pointers: string[] = Object.keys(this._tocPointerMap);
+      for (let i: number = 0, length: number = pointers.length; i < length; i++) {
+        if (itemName.indexOf(pointers[i]) !== -1) {
+          // null assertion used because when pointer map was created we checked for presence of empty `items` array
+          this._tocPointerMap[pointers[i]].items!.push(tocItem);
+          filtered = true;
+          break;
+        }
+      }
+    }
+
+    // If item still not filtered and a `catchAllCategory` config provided push it to it.
+    if (!filtered && this._catchAllPointer && this._catchAllPointer.items) {
+      this._catchAllPointer.items.push(tocItem);
+    }
+  }
+
   // This is a direct copy of a @docCategory inline tag finder in office-ui-fabric-react,
   // but is generic enough to be used for any inline tag
   private _findInlineTagByName(tagName: string, docComment: DocComment | undefined): DocInlineTag | undefined {
+    const tagNameToCheck: string = `@${tagName}`;
```

Comment:
Good change. Are we still planning to take a dependency on api-docs and import `findInlineTagByName` from there?

Reply:
Not sure if we need to take a dependency on `api-docs` for this small utility helper function. `api-docs` is kinda a OUFR specific package and `api-documenter` should be generic. Perhaps the opposite way would make more sense. Have this extracted as an utility function maybe to `api-extractor-model` package and make it available to both places.

Reply:
Ah right, I remember talking about that. That makes sense to me!